### DISCUSSION TITLE
ENOENT: no such file or directory

### DISCUSSION THREAD

**Discussion by EmDash00** [lumen-oss/rocks.nvim#548] [Q&A] [answered]
Trying to use the install script to get a fresh install of rocks.nvim, but I can't seem to get it to work.

```
Installing rocks.nvim...
E5108: Error executing lua: .../bin/nvim-linux64/share/nvim/runtime/lua/vim/_system.lua:244: ENOENT: no such file or directory
stack traceback:
        [C]: in function 'error'
        .../bin/nvim-linux64/share/nvim/runtime/lua/vim/_system.lua:244: in function 'spawn'
        .../bin/nvim-linux64/share/nvim/runtime/lua/vim/_system.lua:335: in function 'system'
        /tmp/nvim.ember/RCSF0G/0.lua:399: in function </tmp/nvim.ember/RCSF0G/0.lua:345>
```

Can someone help me figure out what's going on? I was on the old nightly release but decided to just do a fresh install. I deleted my old rocks directory and reran the script and got this odd message. What seems to be the problem?

**Comment by mrcjkb** [comment]
Hey :wave: 

When `vim.system` errors with `ENOENT: no such file or directory`, it typically means that the command being executed could not be found on the `PATH`.
I've just updated the installer and bootstrap scripts to output which command failed: 

- https://github.com/nvim-neorocks/rocks.nvim/pull/549

**Reply by EmDash00** [reply]
Hi thanks for the swift response. I was previously using a system Luarocks, but I started having issues with it and just decided to let rocks.nvim take care of installing Luarocks. 

I ran it again and the error is about what I suspected. I looked through the source originally and suspected this. It seems to download Luarocks and build it, but when attempting to run it from the temporary directory it was built in, it can't seem to find it. Very odd. Any ideas what could be causing this?

```
stderr: Failed to execute:
/run/user/1000/luarocks-CB511A84450400/bin/luarocks --lua-version=5.1 --tree=/home/ember/.local/share/nvi
m/rocks --server='https://nvim-neorocks.github.io/rocks-binaries/' install rocks.nvim
.../bin/nvim-linux64/share/nvim/runtime/lua/vim/_system.lua:244: ENOENT: no such file or directory
stdout: nil
```

**Reply by mrcjkb** [reply]
Which system are you on? 
I've only encountered this on Windows (in a GitHub actions workflow) so far and haven't been able to solve it yet. 
I think it could be one of the following: 

- The `luarocks` binary is not executable (maybe due to something going wrong during the build)
- luarocks could be trying to call another executable that can't be found on the `PATH` (perhaps `lua`?)
- A permissions issue 

Which issues were you having with the system luarocks? As long as you have Lua 5.1/luajit headers installed, it should be possible to bootstrap rocks.nvim with the system luarocks (on Linux/mac). It should install luarocks 3.11.1 as a dependency and then use that instead of the system luarocks, as long as you don't set the `luarocks_binary` option in `vim.g.rocks_nvim`.

**Reply by EmDash00** [reply, marked as answer]
I found the issue. It's a bug in my `.zshrc`. I added a directory to `PATH` which contained a script named "install" which took precedence over the `install` command. Removing that directory from `PATH` seems to have solved the problem. Thanks for including in the new debugging information. Was helpful to find out what was going on.
